relay-server/src/services/buffer/envelope_store/sqlite.rs

```diff
@@ -370,9 +372,12 @@ impl SqliteEnvelopeStore {
         }
 
         // We sort envelopes by `received_at`.
+        //
         // Unfortunately we have to do this because SQLite `DELETE` with `RETURNING` doesn't
         // return deleted rows in a specific order.
-        extracted_envelopes.sort_by_key(|a| a.received_at());
+        extracted_envelopes.sort_by_key(|a| {
+            Reverse(UnixTimestamp::from_datetime(a.received_at()).unwrap_or(UnixTimestamp::now()))
```

Comment:
I forgot to return the dates in reverse sorted order to honor the API. This is now fixed, the envelopes are returning in descending timestamp order.